modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeLaws.kt

```diff
@@ -50,12 +51,17 @@ object ApplicativeLaws {
     }
 
   fun <F> Applicative<F>.cartesianBuilderMap(EQ: Eq<Kind<F, Int>>): Unit =
-    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int, d: Int, e: Int, f: Int ->
-      mapN(just(a), just(b), just(c), just(d), just(e), just(f)) { (x, y, z, u, v, w) -> x + y + z - u - v - w }.equalUnderTheLaw(just(a + b + c - d - e - f), EQ)
+    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int ->
+      mapN(just(a), just(b), just(c)) { (x, y, z) -> x + y + z }.equalUnderTheLaw(just(a + b + c), EQ)
     }
 
-  fun <F> Applicative<F>.cartesianBuilderTupled(EQ: Eq<Kind<F, Int>>): Unit =
-    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int, d: Int, e: Int, f: Int ->
-      tupledN(just(a), just(b), just(c), just(d), just(e), just(f)).map { (x, y, z, u, v, w) -> x + y + z - u - v - w }.equalUnderTheLaw(just(a + b + c - d - e - f), EQ)
+  fun <F> Applicative<F>.cartesianBuilderTupled2(EQ: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int ->
+      tupledN(just(a), just(b)).map { (x, y) -> x + y }.equalUnderTheLaw(just(a + b), EQ)
+    }
+
+  fun <F> Applicative<F>.cartesianBuilderTupled3(EQ: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int ->
+      tupledN(just(a), just(b), just(c)).map { (x, y, z) -> x + y + z }.equalUnderTheLaw(just(a + b + c), EQ)
```

Comment:
We should use `Eq<Tuple3<Int, Int, Int>>` here because `Int` + `Int` + `Int` might result in weird results.

Reply:
We should have a general rule that using commutative monoids should not be allowed at all in those kind of tests. It fails to uncover trivial ordering errors.

Reply:
Yes! I've been guilty of this before too.